src/api/XRSession.js

```diff
@@ -77,17 +81,10 @@ export default class XRSession extends EventTarget {
       activeRenderState: initialRenderState,
       pendingRenderState: null,
       viewerSpace: new XRReferenceSpace("viewer"),
-      viewSpaces: [],
+      get viewSpaces() { return device.getViewSpaces(mode) || defaultViewSpaces; },
```

Comment:
I was going to make a change that would have getViewSpaces return an array of XREye instead of an array of XRSpace, but decided not to as it makes avoiding garbage here difficult.